vars/withOpenshiftServices.groovy

```diff
@@ -21,21 +21,39 @@ def cleanup(names) {
     }
 }
 
-def createOpenshiftResources(services, names) {
-    Map jobs = [:]
+def createOpenshiftResources(List<String> services, List<String> names) {
+    Map<String, Closure> jobs = [:]
     for (int i = 0; i < services.size(); i++) {
         String service = services[i]
         String name = names[i]
         jobs[name] = {
             openshiftCreateResource(getDeploymentConfigYaml(service, name))
             openshiftCreateResource(getServiceYaml(service, name))
             openshiftScale deploymentConfig: name,  replicaCount: 1, verifyReplicaCount: 1, waitTime: 600000
+            waitForServiceToBeReady service, name
         }
     }
     parallel jobs
 }
 
-String sanitizeObjectName(s) {
+def waitForServiceToBeReady(String service, String name) {
+    if (service == 'mongodb') {
+        writeFile(
+            file: 'checkMongo.sh',
+            text: "echo 'daves\ndaves()\n/' | curl -v telnet://${name}:27017/"
```

Comment:
Don't ask about the daves, it's sekrit codes 🦀 

<details>
I accidentally stumbled across this (well, something similar: I had 'close' instead of 'daves') when trying to get the curled telnet connection to close successfully. It seems to be the number of bytes that MongoDB needs to realise that we're not speaking it's protocol and cleanly closes the connection. At least I think that's what's happening.
</details>